### DISCUSSION TITLE
Scalar coordinate inheritence between dataset and dataarray

### DISCUSSION THREAD

**Discussion by rcaneill** [pydata/xarray#6340] [Q&A] [answered]
I have a question, which I don't know if this is a bug, a feature, or something I'm doing wrong (I hope it was correct to open a discussion and not an issue).

## Use case

We are developing a small wrapper around the gsw (Gibbs Sea Water) package that is used to compute thermodynamic properties of the ocean (e.g. density, potential temperature, etc). The goal of this wrapper is to add a xarray layer around the gsw package, adding when possible standard_name attribute, units, etc. For some variables, the CF-convention tells that a scalar coordinate should be added, counting for the reference pressure used for the calculation, e.g. `sea_water_potential_density`, that is described this way: `Sea water potential density is the density a parcel of sea water would have if moved adiabatically to a reference pressure, by default assumed to be sea level pressure. To specify the reference pressure to which the quantity applies, provide a scalar coordinate variable with standard name reference_pressure. The density of a substance is its mass per unit volume. For sea water potential density, if 1000 kg m-3 is subtracted, the standard name sea_water_sigma_theta should be chosen instead.`

So the main question is: how to attach a scalar coordinate to only one dataarray and not the whole dataset?

## Code

If the coordinate is added to a dataarray in a dataset, it seems to be only attach to this specific dataarray:

```python
import xarray as xr
ds = xr.Dataset()
temp = xr.DataArray([[0,1,2], [3,4,5]], dims=('lat', 'lon'))
sigma1 = xr.DataArray([25.0], dims=('depth'))
ds['temp'] = temp
ds['sigma1'] = sigma1
print(ds, '\n')
ds.sigma1.coords['pref'] = 1000
print(ds)
```
Output:
```
<xarray.Dataset>
Dimensions:  (depth: 1, lat: 2, lon: 3)
Dimensions without coordinates: depth, lat, lon
Data variables:
    temp     (lat, lon) int64 0 1 2 3 4 5
    sigma1   (depth) float64 25.0 

<xarray.Dataset>
Dimensions:  (depth: 1, lat: 2, lon: 3)
Dimensions without coordinates: depth, lat, lon
Data variables:
    temp     (lat, lon) int64 0 1 2 3 4 5
    sigma1   (depth) float64 25.0

```

However, if the scalar coordinate is added to a dataarray that is merge into a dataset, it becomes a coordinate of the dataset:
```python
import xarray as xr
ds = xr.Dataset()
temp = xr.DataArray([[0,1,2], [3,4,5]], dims=('lat', 'lon'))
sigma1 = xr.DataArray([25.0], dims=('depth'))
sigma1.coords['pref'] = 1000

ds['temp'] = temp
ds['sigma1'] = sigma1

print(ds)
```

Output
```
<xarray.Dataset>
Dimensions:  (depth: 1, lat: 2, lon: 3)
Coordinates:
    pref     int64 1000
Dimensions without coordinates: depth, lat, lon
Data variables:
    temp     (lat, lon) int64 0 1 2 3 4 5
    sigma1   (depth) float64 25.0
```

## What we tried

The CF convention describes [scalar coordinate](https://cfconventions.org/Data/cf-conventions/cf-conventions-1.8/cf-conventions.html#scalar-coordinate-variables) through the attribute `coordinates`, but this does not seem to be decoded by xarray (in my tests).

Anyone would have any idea if this is possible within the xarray model?

## Other links
[Link to the issue in the gsw wrapper gsw-xarray](https://github.com/DocOtak/gsw-xarray/issues/32)

**Comment by dcherian** [comment, marked as answer]
As you've found, xarray does not support  this. `cf_xarray`, on the other hand, does parse the `coordinates` attribute

``` python
import cf_xarray
import xarray as xr

ds = xr.Dataset()
temp = xr.DataArray([[0, 1, 2], [3, 4, 5]], dims=("lat", "lon"))
sigma1 = xr.DataArray(
    [25.0], dims=("depth"), coords={"pref": 1000}, attrs={"coordinates": "pref"}
)

ds["temp"] = temp
ds["sigma1"] = sigma1

print(ds.cf["temp"].coords)
print(ds.cf["sigma1"].coords)
```

```
Coordinates:
    *empty*
Coordinates:
    pref     int64 1000
```

**Reply by rcaneill** [reply]
Thanks for answering quickly!
